src/protocols/secure_channel/CASESession.cpp

```diff
@@ -430,38 +389,42 @@ CHIP_ERROR CASESession::HandleSigmaR1(System::PacketBufferHandle & msg)
     err = tlvReader.EnterContainer(containerType);
     SuccessOrExit(err);
 
-    err = tlvReader.FindElementWithTag(CASETLVTag::kSessionID, suppTlvReader);
+    err = tlvReader.FindElementWithTag(TLV::ContextTag(1), suppTlvReader);
```

Comment:
The spec expect tag-order. This reads in any order. There should be a utility method to "ensure at least N context tags in order starting at tag M", as a pre-validation of schema. Then `FindElementWithTag` is fine once that is validated.